internal/usecase/interactor/user.go

```diff
@@ -241,6 +169,109 @@ func (i *User) Signup(ctx context.Context, inp interfaces.SignupParam) (u *user.
 	return u, team, nil
 }
 
+func (i *User) authSystemSignup(ctx context.Context, inp interfaces.SignupParam, tx repo.Tx) (repo.Tx, string, string, *user.User, *user.Team, error) {
+	if *inp.Name == "" {
+		return nil, "", "", nil, nil, interfaces.ErrSignupInvalidName
+	}
+	if _, err := mail.ParseAddress(*inp.Email); err != nil {
+		return nil, "", "", nil, nil, interfaces.ErrInvalidUserEmail
+	}
+	if *inp.Password == "" {
+		return nil, "", "", nil, nil, interfaces.ErrSignupInvalidPassword
+	}
```

Comment:
Move also this validation out of the function to move codes to start transaction.